### PR TITLE
Add aeval: Asynchronous Equivalent to eval in aioconsole

### PR DESCRIPTION
**Pull request by Akm0d**
This PR introduces a new `aeval` function to the `aioconsole` library, providing an asynchronous equivalent to Python's built-in `eval`. The `aeval` function enhances `aioconsole` by enabling users to evaluate expressions asynchronously within an interactive console environment, complementing the existing `aexec` function.

### Key Features of `aeval`:

**Asynchronous Evaluation:**
- The `aeval` function allows for the evaluation of expressions that may involve asynchronous operations. This is particularly useful in environments where asynchronous code is prevalent, such as web servers or other I/O-bound applications.

**Behavior Consistent with `eval`:**
- `aeval` functions in a manner consistent with Python’s built-in `eval`. It evaluates single-line expressions and raises the same exceptions as `eval` for invalid input.
- Notably, `aeval` does **not** handle multi-line inputs or statements (e.g., function definitions, loops), maintaining strict compatibility with the `eval` behavior which is limited to single expressions.

**Error Handling:**
- The function gracefully handles syntax errors and other exceptions, providing clear feedback when issues arise during the evaluation of expressions, just as `eval` does.

**Seamless Integration with Existing `aexec`:**
- The implementation of `aeval` builds upon the `aexec` function, ensuring consistent behavior and making use of the existing compilation and execution mechanisms within `aioconsole`.

### Example Usage:

```python
import asyncio
from aioconsole import aeval

async def main():
    local_namespace = {"x": 10}
    result = await aeval("x + 5", local_namespace)
    print(result)  # Outputs: 15

asyncio.run(main())
```

This PR ensures that `aeval` behaves exactly like the built-in `eval`, with the added capability of handling asynchronous functions, making it a powerful tool for asynchronous interactive environments.

### CODE REVIEW

**Comment by vxgmichel** [comment]
Hi @Akm0d and thanks for taking the time to make this PR.

I'm not against introducing an `aeval` function, but in this case I would like it to behave similarly to the builtin `eval`.

For instance `eval("def foo(): return 42")` raises a `SyntaxError` but your implementation does not.

Moreover, `eval("x = 1\nx + 1")` also raises a `SyntaxError` but your implementation returns `2`.

As far as I remember, implementing `aeval` was not trivial which is why I did not take the time to do it in the first place :)

**Comment by Akm0d** [comment]
thanks for taking a look so quickly!  i'll do some fone tuning to resolve what you pointed out

**Comment by Akm0d** [comment]
@vxgmichel I updated the code to ensure that `aeval` functions exactly the same as the builtin `eval` (except that it handles async expressions of course).

I also modified my  tests to compare the results of `aeval` and `eval` in a wide variety of edge cases.

**Comment by Akm0d** [comment]
Ok I've added more test cases to achieve full code coverage.  Also raising the same `TypeError` as `eval` when `local` is not a dict.

File: aioconsole/execute.py [hunk 1]
```diff
@@ -148,3 +148,31 @@ async def aexec(source, local=None, stream=None, filename="<aexec>"):
...
+    """Asynchronous equivalent to *eval*."""
+    if local is None:
+        local = {}
+
+    if not isinstance(local, dict):
+        raise TypeError("globals must be a dict")
+
+    key = "__result__"
+    # Ensure that the result key is unique within the local namespace
+    # while key in local: key += "_"
```
Comment (vxgmichel) [inline aioconsole/execute.py:163]: Why is this commented out?

**Review by vxgmichel — APPROVED** [review]
LGTM :+1:  

I'll have to fix the python 3.12 compatibility though

File: aioconsole/execute.py (hunk 1 above)
Comment (Akm0d) [reply aioconsole/execute.py:163]: I thought it might be overkill 😅 But wanted to leave there for reference.  Are you in favor of remove that line or uncomment it?

Comment (vxgmichel) [reply aioconsole/execute.py]: I decided to keep this logic [54e6c09](https://github.com/vxgmichel/aioconsole/pull/120/commits/54e6c0911733f2993dc33a59fb3758808d408a73)

**Comment by vxgmichel** [comment]
Merged :tada: 
Thanks a lot for the PR @Akm0d

**Comment by vxgmichel** [comment]
Release in [v0.8.0](https://github.com/vxgmichel/aioconsole/releases/tag/v0.8.0) :tada:
